diff --git a/lib/portal/docgenerator/popup.py b/lib/portal/docgenerator/popup.py
index 05babb4..d08263f 100644
--- a/lib/portal/docgenerator/popup.py
+++ b/lib/portal/docgenerator/popup.py
@@ -113,7 +113,7 @@ class Popup(object):
                   </div>
                   <div class="modal-body-container">
                     <div class="modal-body modal-body-error alert alert-danger padding-all-small padding-left-large">
-                      Error happened on the server
+                        <pre>Error happened on the server</pre>
                     </div>
                     <div class="modal-body modal-body-message alert alert-success padding-all-small padding-left-large">
                     </div>
@@ -196,8 +196,14 @@ class Popup(object):
                 },
                 error: function(response, statusText, xhr, $form) {
                     if (response) {
-                        var responsetext = response.responseJSON || response.responseText;
-                        this.popup.find('.modal-body-error').text(responsetext);
+                        var errortext = response.responseJSON || response.responseText;
+                        try {
+                            var eco = JSON.parse(errortext);
+                            errortext = eco.backtrace;
+                        } catch (e) {
+                            // dont do anything just use errortext
+                        }
+                        this.popup.find('.modal-body-error > pre').text(errortext);
                     }
                     if (response && (response.status == 400 || response.status == 409)){
                         this.popup.find("input,select,textarea").prop("disabled", false)
